DQMServices/Core/interface/MonitorElement.h

```diff
@@ -76,13 +96,37 @@ namespace dqm::impl {
     using Scalar = MonitorElementData::Scalar;
     using Kind = MonitorElementData::Kind;
 
+    // Comparison helper used in DQMStore to insert into sets. This needs deep
+    // private access to the MEData, that is why it lives here.
+    struct MEComparison {
+      using is_transparent = int;  // magic marker to allow C++14 heterogeneous set lookup.
+
+      auto make_tuple(MonitorElement *me) const { return std::make_tuple(me->getPathname(), me->getName()); }
+      auto make_tuple(MonitorElementData::Path const &path) const {
+        return std::make_tuple(path.getDirname(), path.getObjectname());
+      }
+      bool operator()(MonitorElement *left, MonitorElement *right) const {
+        return make_tuple(left) < make_tuple(right);
```

Comment:
If I'm not mistaken the `path.getDirname()` and `path.getObjectname()` return `string const&`. As far as I can tell, in this case `std::make_tuple()` seems to return `std::tuple<std::string, std::string>` and thus copies the string objects. I guess from the comment above that these functions are used as comparison functions in set insertion/sorting, and if the number of objects to be inserted/sorted is large, the string copying has a risk of becoming expensive.

By quick test with compiler explorer using `std::reference_wrapper` along
```cpp
  return std::make_tuple(std::reference_wrapper(path.getDirname()), std::reference_wrapper(path.getObjectname()));`
```
seems to avoid string copies (a hand-written two-string comparison might be even more performant).

Reply:
Thanks for the hint! -- my profiling also suggests that there are still string copies around there, but I did not change anything as long as performance was acceptable. Using `reference_wrapper` might be an easy way to fix that.

Reply:
Ah, found the old comment! I use reference_wrapper now.